arm/README.md

```diff
@@ -91,6 +91,58 @@ post,
 [Developer's Guide to Auth with Azure Resource Manager API](http://www.dushyantgill.com/blog/2015/05/23/developers-guide-to-auth-with-azure-resource-manager-api/),
 that is also quite helpful.
 
+### Use an authentication file
+
+This SDK also supports authentication with a JSON file containing credentials for the service principal. In the Azure CLI, you can create a service principal and its authentication file with this command:
+
+``` bash
+az ad sp create-for-rbac --sdk-auth > mycredentials.json
+```
+
+THe file lloks like this, in case you want to create it yourself:
+
+``` json
+{
+    "clientId": "ad735158-65ca-11e7-ba4d-ecb1d756380e",
+    "clientSecret": "b70bb224-65ca-11e7-810c-ecb1d756380e",
+    "subscriptionId": "bfc42d3a-65ca-11e7-95cf-ecb1d756380e",
+    "tenantId": "c81da1d8-65ca-11e7-b1d1-ecb1d756380e",
+    "activeDirectoryEndpointUrl": "https://login.microsoftonline.com",
+    "resourceManagerEndpointUrl": "https://management.azure.com/",
+    "activeDirectoryGraphResourceId": "https://graph.windows.net/",
+    "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
+    "galleryEndpointUrl": "https://gallery.azure.com/",
+    "managementEndpointUrl": "https://management.core.windows.net/"
+}
+```
+
+Save this file in a secure location on your system where your code can read it. Set an environment variable with the full path to the file:
+
+``` bash
+export AZURE_AUTH_LOCATION=/secure/location/mycredentials.json
```

Comment:
I would mention this environment variable higher up as well. It is important that people know that unlike other examples of environment variables in our Samples and README, this one is referenced by our library. 

Also, we should add a PowerShell example as well.

Reply:
Fixed